pkg/networkserver/networkserver.go

```diff
@@ -346,3 +346,16 @@ func (ns *NetworkServer) RegisterHandlers(s *runtime.ServeMux, conn *grpc.Client
 func (ns *NetworkServer) Roles() []ttnpb.PeerInfo_Role {
 	return []ttnpb.PeerInfo_Role{ttnpb.PeerInfo_NETWORK_SERVER}
 }
+
+func (ns *NetworkServer) Close() {
+	ns.Component.Close()
+
+	logger := ns.Logger()
+	for _, cl := range ns.applicationServers {
+		logger.Debug("Close Application Server link")
+		if err := cl.Close(); err != nil {
+			logger.WithError(err).Warn("Failed to close AS link")
```

Comment:
1. Consistent component naming, `Application Server`
2. `else` for `Debug` log below